force-app/main/default/classes/SOQL.cls

```diff
@@ -313,19 +456,26 @@ public with sharing class SOQL {
      * @param List of cloned Opportunity Ids
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
-        // Query the original Opportunity Line Items
-
-        // Create a list to hold the new line items
-
-        // Generate a random number between 1 and 3
-
-        // Loop through each cloned Opportunity Id
-
-            // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
-
-        // Insert the new line items
+        List<OpportunityLineItem> olis = new List<OpportunityLineItem>([
+            SELECT Id, Quantity, PricebookEntryId, UnitPrice
+            FROM OpportunityLineItem
+            WHERE OpportunityId =:origOppId WITH SECURITY_ENFORCED
+        ]);
+    
+        List<OpportunityLineItem> olisToInsert = new List<OpportunityLineItem>();
+        for (Id id : cloneOppIds) {
+            for (OpportunityLineItem oli : olis) {
+                OpportunityLineItem newOli = oli.clone(false, true);
+                Integer numOli = (Integer)Math.floor(Math.random() * 10) + 1;
```

Comment:
I would move this Integer outside your for loop so that it isn't calculating each iteration